Route Details: Conditions table

dtaylor113 opened the pull request:
On Route Details page replaced inline route status warning messages with a Conditions table to match what was done in #97.

**Before:**
![image](https://user-images.githubusercontent.com/12733153/41548978-68f29c5e-72f2-11e8-8c40-a86547e967c9.png)

**After:**
![image](https://user-images.githubusercontent.com/12733153/41548853-1844df10-72f2-11e8-9ad9-d403d666e200.png)
_New  Router Section_
![image](https://user-images.githubusercontent.com/12733153/41779047-8f37f13e-75fe-11e8-9510-f931667afaa0.png)

@openshift/team-ux-review

spadgett commented on frontend/public/components/routes.tsx:
```diff
@@ -295,6 +249,11 @@ const RouteDetails: React.SFC<RoutesDetailsProps> = ({obj: route}) => <React.Fra
   </div>
+  <div className="co-m-pane__body">
+    <Conditions obj={(route.status && route.status.ingress)
+      ? _.flatten(route.status.ingress.map((ingress)=>ingress.conditions))
```
We should have separate tables for each ingress instead of flattening. You need to know which hostname each of the conditions are for, so the hostname should be in the header above each table.

dtaylor113 replied:
So would it be something like 
"www.daves-great-app.com" or
"www.daves-great-app.com Conditions" or 
"Conditions for www.daves-great-app.com"
as the table title?  Or have the hostname as a sub-heading, something like:
"Conditions
www.daves-great-app.com  <-- smaller text
---table---"

spadgett replied:
Looking at the YAML, the other items are important, too.

```yaml
    - host: console-tectonic-console.svc.ci.openshift.org
      routerName: router
      conditions:
        - type: Admitted
          status: 'True'
          lastTransitionTime: '2018-05-23T16:28:52Z'
      wildcardPolicy: None
```

I would suggest something like


console-tectonic-console.svc.ci.openshift.org (header)

Router: router
Wildcard Policy: None

(conditions table)

dtaylor113 replied:
Hi, I don't think I've seen any 'label: value' side-by-sides in the UI, it's always label ontop of value.  This is what I have so far:
![image](https://user-images.githubusercontent.com/12733153/41557350-5ee9e14a-730b-11e8-93d5-55f5f5fe18f6.png)
- It seems like we're missing a table title still, something like 'Conditions'

dtaylor113 replied:
I do see this in nodes, should I do the same stying for Router & Wildcard Policy?
![image](https://user-images.githubusercontent.com/12733153/41558204-13991c58-730e-11e8-91c1-20d6ad527871.png)
Meaning grey for the label, and black for the value, side-by-side?  -thanks

dtaylor113 replied:
cc @beanh66  :-)

spadgett replied:
I would use the existing style. I'd probably add "Conditions" above the table as well

spadgett commented on frontend/public/components/conditions.tsx:
```diff
@@ -40,6 +42,6 @@ export const Conditions: React.SFC<ConditionsProps> = ({obj}) => {
 /* eslint-disable no-undef */
 export type ConditionsProps = {
-  obj: any,
+  obj: any[],
```
I'd rather call this property `conditions` instead of `obj`. Might be better to call the component `ConditionsTable`.

dtaylor113 replied:
Not quite sure how to do that.  If I replace `obj` with `conditions` here and in the component header `export const Conditions: React.SFC<ConditionsProps> = ({conditions})...`, I get
```
ERROR in /home/dtaylor/repos/openshift/origin/src/github.com/openshift/console/frontend/public/components/routes.tsx(253,17):
TS2322: Type '{ obj: {}[]; }' is not assignable to type 'IntrinsicAttributes & ConditionsProps & { children?: ReactNode; }'.
  Type '{ obj: {}[]; }' is not assignable to type 'ConditionsProps'.
    Property 'conditions' is missing in type '{ obj: {}[]; }'.
```

spadgett replied:
Looks like you didn't update where the component is used in routes.tsx line 253

benjaminapetersen replied:
Can we be more specific than `obj: any[]`?

For example, in existing code I've seen types like:

```TypeScript  
allkinds: {[s: string]: K8sKind},
// 
match: RouterMatch<{ns: string, plural: string, appName?: string}>;
//
flags: {[name: string]: boolean};
//
maxValues: {[_: string]: number},
totals: {[_: string]: number}
```
To drop a few examples.  If we can capture the shape, even minimally better than `any`, should we do so?

dtaylor113 replied:
ok, I got it working now.  @benjaminapetersen what does the _ signify in `{[_: string]: number}`? -thanks

benjaminapetersen replied:
I have no idea. 😆 
Generally speaking, I see complex objects being represented via types by making assertions on some of the key properties.  I think that's the main takeaway.   Dabbling with this myself as well.

benjaminapetersen replied:
Could be a Q for @alecmerdler, if he has any advice on the best way to make the most of typescript in these cases.

benjaminapetersen replied:
Pondering the difference between types like these:

https://github.com/openshift/console/blob/master/frontend/public/components/cloud-services/index.tsx#L85

And then types that are `props` for Components.  Not sure if there is a dividing line between how specific we should be with `prop` types vs other types.  

Thinking out loud.

spadgett commented on frontend/public/components/conditions.tsx:
```diff
@@ -1,8 +1,10 @@
-export const Conditions: React.SFC<ConditionsProps> = ({obj}) => {
-  const conditions = _.get(obj, 'status.conditions');
+export const Conditions: React.SFC<ConditionsProps> = ({obj: conditions}) => {
+  if (!conditions.length) {
```
Missing conditions should be handled below since we display "No conditions." If we want an empty array check, we should use `_.isEmpty()` below.

spadgett commented:
We should update the conditions table message column to use `white-space: pre-line`

benjaminapetersen commented on frontend/__tests__/components/route-pages.spec.tsx:
```diff
@@ -242,69 +242,3 @@ describe(RouteStatus.displayName, () => {
   });
 });
-
-describe(RouteWarnings.displayName, () => {
```
Replacing these tests with other tests? Or a straight elimination. (I've personally added no tests yet, def want to fix that)

dtaylor113 replied:
We removed/replaced these calculated/parsed warnings with the Conditions table which just shows the raw data/warnings.

beanh66 commented:
@dtaylor113 I have the same question for @spadgett... do we need the conditions table when the status is accepted?

spadgett commented:
> do we need the conditions table when the status is accepted?

Transition time would be nice to know... Maybe we can add that. I would also put a `-` when no reason or message exists.

Route conditions are a little different. Generally those fields aren't empty.

jhadvig commented on frontend/public/components/hpa.tsx:
```diff
@@ -174,7 +174,7 @@ export const HorizontalPodAutoscalersDetails: React.SFC<HorizontalPodAutoscalers
   </div>
   <div className="co-m-pane__body">
-    <Conditions obj={hpa} />
+    <Conditions obj={_.get(hpa, 'status.conditions')} />
```
Looks like the `status.conditions` is always present on the HPA, in that case we dont need `_.get`

dtaylor113 commented:
Hi @spadgett, @beanh66, and @benjaminapetersen, I updated the screenshots in the PR's Description above.  I believe I addressed all comments including @benjaminapetersen request for more TypeScripting.  
- Thanks :-)

stevekuznetsov commented:
/test backend
/test frontend

spadgett commented on frontend/public/components/conditions.tsx:
```diff
@@ -11,15 +10,17 @@ export const Conditions: React.SFC<ConditionsProps> = ({obj}) => {
     </div>
     <div className="col-xs-3 col-sm-3">
-      {condition.reason}
+      {condition.reason ? condition.reason : '-'}
```
```jsx
{condition.reason || '-'}
```

dtaylor113 replied:
Fixed

spadgett replied:
{condition.reason || '-'}

spadgett commented on frontend/public/components/conditions.tsx:
```diff
@@ -11,15 +10,17 @@ export const Conditions: React.SFC<ConditionsProps> = ({obj}) => {
-    <div className="col-xs-3 col-sm-5">
-      {condition.message}
+    <div className="col-xs-3 col-sm-5 co-pre-line-wrap">
+      {/* remove initial newline which appears in route messages */}
```
Are you sure the newline isn't because you have a newline after the div? Can you try...

```jsx
<div className="col-xs-3 col-sm-5 co-pre-line-wrap">{condition.message || '-'}</div>
```

dtaylor113 replied:
ok, I'll try, but I'm seeing it in the YAML as well:
![image](https://user-images.githubusercontent.com/12733153/41662289-c5fa7df4-746e-11e8-93d6-fb7adbc20cee.png)

dtaylor113 replied:
Hi, tried 
```jsx
<div className="col-xs-3 col-sm-5 co-pre-wrap">{condition.message || '-'}</div>
```
Still seeing an initial newline

spadgett replied:
I think _.trim with pre-line should be enough to format this without the regex

spadgett commented on frontend/public/components/conditions.tsx:
```diff
@@ -11,15 +10,17 @@ export const Conditions: React.SFC<ConditionsProps> = ({obj}) => {
   return <React.Fragment>
-    <h1 className="co-section-title">Conditions</h1>
+    {title && <h1 className="co-section-title">{title}</h1>}
+    {subTitle && <h4 style={{paddingTop: 4}}>{subTitle}</h4>}
```
Let's not use inline styles. (Should this have a unit?)

spadgett replied:
I'd rather just remove the title from this component and make it only the table instead of having title / subTitle props

spadgett commented on frontend/public/components/routes.tsx:
```diff
@@ -267,6 +222,25 @@ const RouteTargetRow = ({route, target}) => <tr>
   <td>{calcTrafficPercentage(target.weight, route)}</td>
 </tr>;
 
+const IngressDetails: React.SFC<IngressDetailsProps> = ({ingresses}) => <React.Fragment>
```
You shouldn't need the outer React.Fragment. It can just be

```jsx
const IngressDetails: React.SFC<IngressDetailsProps> = ({ingresses}) => _.map(ingresses, ...);
```

spadgett replied:
I'd call this `RouteIngressStatus`

dtaylor113 replied:
When I do: `const IngressDetails: React.SFC<IngressDetailsProps> = ({ingresses}) => _.map(..`, I get:
```
ERROR in /home/dtaylor/repos/openshift/origin/src/github.com/openshift/console/frontend/public/components/routes.tsx(225,7):
TS2322: Type '({ ingresses }: RouteIngressStatusProps & { children?: ReactNode; }) => Element[]' is not assignable to type 'StatelessComponent<RouteIngressStatusProps>'.
  Type 'Element[]' is not assignable to type 'ReactElement<any>'.
    Property 'type' is missing in type 'Element[]'.
```
Component:
```
const RouteIngressStatus: React.SFC<RouteIngressStatusProps> = ({ingresses}) => _.map(ingresses, (ingress, i) =>
  <React.Fragment key={i}>
    <h1 className="co-section-title">{ingress.host}</h1>
    <dl>
      <dt>Router</dt>
      <dd>{ingress.routerName}</dd>
      <dt>Wildcard Policy</dt>
      <dd>{ingress.wildcardPolicy}</dd>
    </dl>
    <h4 className="co-m-route-ingress-conditions___title">Conditions</h4>
    <Conditions key={ingress.host} conditions={ingress.conditions} />
  </React.Fragment>);
```
I think because whatever is returned needs to have a single root parent element; the `_.map` returns an array of children (multiple ingress status tables...if they exist)

spadgett commented on frontend/public/components/routes.tsx:
```diff
@@ -267,6 +222,25 @@ const RouteTargetRow = ({route, target}) => <tr>
+const IngressDetails: React.SFC<IngressDetailsProps> = ({ingresses}) => <React.Fragment>
+  {_.map(ingresses, (ingress, i) => <React.Fragment key={i}>
+    <h1 className="co-section-title">{ingress.host}</h1>
+    <table className="co-m-route-horizontal-props">
```
Use a `dl` instead of a `table`

dtaylor113 replied:
Hi, a little confused :-) The default `dl, dt, dd` defined in `_common.scss & _base.scss` make the `dt` uppercase and ontop of the `dd`, like we see in the Detail pages.  I thought we wanted: 'label: value', side-by-side?  I've only seen one example of this, but it used `<spans>'s` and only matched up due to the labels being the same number of chars.

spadgett replied:
I think that the existing dl, dt, dd styles are fine.

spadgett commented on frontend/public/components/routes.tsx:
```diff
@@ -295,6 +268,9 @@ const RouteDetails: React.SFC<RoutesDetailsProps> = ({obj: route}) => <React.Fra
       </div>
     </div>
   </div>
+  {(route.status && route.status.ingress) && <div className="co-m-pane__body">
```
Maybe something like

```jsx
{_.isEmpty(route.status.ingress)
  ? <div className="cos-status-box">
      <div className="text-center">No Route Status</div>
    </div>
  : <IngressDetails ingresses={route.status.ingress} />}
```

nit: please put a space before `/` in self-closing tags throughout for consistency with existing code

spadgett commented on frontend/public/components/routes.tsx:
```diff
@@ -130,52 +131,6 @@ export const RouteStatus: React.SFC<RouteStatusProps> = ({obj: route}) => {
-  addTLSWarnings(route, warnings);
-  addIngressWarnings(route, warnings);
-  return <React.Fragment>
-    {warnings.map((warning, index) => <div key={index} className="co-m-route-warnings">{warning}</div>)}
```
Can you remove this class if it's no longer used?

dtaylor113 replied:
removed

benjaminapetersen replied:
Our linting checks this correct?  If not, we should look into that, would rather have tooling tell us what is/is not used.

spadgett commented on frontend/public/style/_common.scss:
```diff
@@ -149,6 +149,10 @@
   white-space: pre-wrap;
 }
 
+.co-pre-line-wrap {
```
We should call this just `.co-pre-line`, although it might be better to just use `.co-pre-wrap` for the table

beanh66 commented:
@dtaylor113 Thanks this is looking good. Would you mind adding a screenshot of the rejected case that shows the full page so we can see this section more in context?

dtaylor113 commented:
> @dtaylor113 Would you mind adding a screenshot of the rejected case that shows the full page so we can see this section more in context?

Sure, here:
![image](https://user-images.githubusercontent.com/12733153/41662190-8bb161f8-746e-11e8-8f53-68bb6aff0673.png)

serenamarie125 commented:
@dtaylor113 @spadgett typically Conditions is at a higher level heading.  On this page, is Conditions intended to be a secondary heading under "www.davlestlsroute.com"?

cshinn commented:
@dtaylor113 the "Router" and "wildcard policy" labels should be styled similar to "termination type" and the others:  No colon afterwards, all caps, and stacked with the value rather than inline.

spadgett commented:
> typically Conditions is at a higher level heading. On this page, is Conditions intended to be a secondary heading under "www.davlestlsroute.com"?

There are separate conditions for each hostname. (Hostname can be changed.)

dtaylor113 commented:
> @dtaylor113 the "Router" and "wildcard policy" labels should be styled similar to "termination type" and the others: No colon afterwards, all caps, and stacked with the value rather than inline.

Hi, yes, sorry, there was a misunderstanding on how we wanted to show these fields.  I'm switching back to the default.

spadgett commented:
> There are separate conditions for each hostname. (Hostname can be changed.)

We could do something like change the header to "Ingress Status" and make the hostname another field we display.

dtaylor113 commented:
Hi, I believe I have addressed all of the coding comments.
I understand the design is still under review.
I updated this PR's Description with the latest screenshots
-Thanks

spadgett commented on frontend/public/components/routes.tsx:
```diff
@@ -267,6 +222,22 @@ const RouteTargetRow = ({route, target}) => <tr>
+const RouteIngressStatus: React.SFC<RouteIngressStatusProps> = ({ingresses}) =>
+  <React.Fragment>
+    {_.map(ingresses, (ingress, i) =>
+      <React.Fragment key={i}>
```
key={ingress.host}

spadgett commented on frontend/public/components/routes.tsx:
```diff
@@ -267,6 +222,22 @@ const RouteTargetRow = ({route, target}) => <tr>
+          <dd>{ingress.wildcardPolicy}</dd>
+        </dl>
+        <h4 className="co-m-route-ingress-conditions___title">Conditions</h4>
+        <Conditions key={ingress.host} conditions={ingress.conditions} />
```
key shouldn't be needed here?

dtaylor113 commented:
@serenamarie125, @beanh66 , @cshinn,  to me if feels like the line spacing is off between the various content in this section:
![image](https://user-images.githubusercontent.com/12733153/41673680-9c65fa32-748b-11e8-9b36-19211c7cc320.png)

Note the above screenshot is showing a route ingress status table for one hostname ("www.davestlsroute.com"), this would be repeated if there where multiple hostnames

I think this is what @spadgett suggested in an earlier comment:
![image](https://user-images.githubusercontent.com/12733153/41673985-9efcd698-748c-11e8-83e6-22166e63d311.png)

dtaylor113 commented:
Addressed last 3 comments from @spadgett.

Adjusted spacing so that 'Conditions' match spacing of other fields/values:
![image](https://user-images.githubusercontent.com/12733153/41675771-993fefec-7491-11e8-8339-9695d5bc43d2.png)

spadgett commented on frontend/public/components/routes.tsx:
```diff
@@ -267,6 +222,22 @@ const RouteTargetRow = ({route, target}) => <tr>
+          <dt>Wildcard Policy</dt>
+          <dd>{ingress.wildcardPolicy}</dd>
+        </dl>
+        <h4 className="co-m-route-ingress-conditions___title">Conditions</h4>
```
Really this should be an `h2` semantically. I wonder if that adds more margin.

Not necessary to change in this PR

spadgett commented on frontend/public/style/_common.scss:
```diff
@@ -145,6 +145,10 @@
   border-radius: 2px;
 }
 
+.co-pre-line-wrap {
```
.co-pre-line, otherwise it's not clear if it's pre-line or pre-wrap since both are in the name.

dtaylor113 replied:
Fixed these last comments -thanks!

spadgett commented on frontend/public/components/conditions.tsx:
```diff
@@ -11,15 +10,15 @@ export const Conditions: React.SFC<ConditionsProps> = ({obj}) => {
-      {condition.message}
+    {/* remove initial newline which appears in route messages */}
+    <div className="col-xs-3 col-sm-5 co-pre-line">
+      {condition.message ? condition.message.replace(/^.*?\r?\n/, '') : '-'}
```
This should use _.trim instead of the regex

```
{_.trim(condition.message) || '-'}
```

dtaylor113 replied:
thanks, fixed

dtaylor113 commented:
@serenamarie125 

> can you check the spacing? I think the spacing under the WILDCARD/None pair should be increased to match the spacing above it.

Hi, the spacing under the ROUTER/router is `margin-bottom: 20px`.  Because WILDCARD/None is the last child of the list it has a `margin-bottom: 0` but I added a `margin-top: 20px` of the Conditions label to match the spacing.

I'm wondering if I should just include the Conditions label and table in the list so it matches better:  Here's an example:
![image](https://user-images.githubusercontent.com/12733153/41723745-dfe234ac-7539-11e8-9aae-4eb6753009f4.png)
Note Conditions is now CONDITIONS.

> if you are trying to be consistent with #97, the first table in your example seems to match, but the widths of the other columns in the other 2 tables are not consistent. It looks like we should decrease the width of Reason and increase the width of Message

Actually, both this PR and #97 use the same Conditions table component.  The screenshots above were taken at different screen sizes and the content of the tables differ, so that is why they look different.

beanh66 commented:
@dtaylor113 I like that idea of not using the actual route name as a header. I am mocking something up for you. Quick question - what happens to the status when there are multiple routes? Can there be multiple statuses or just accepted or rejected?

dtaylor113 commented:
> @dtaylor113 I like that idea of not using the actual route name as a header. I am mocking something up for you. Quick question - what happens to the status when there are multiple routes? Can there be multiple statuses or just accepted or rejected?

@beanh66, I 'think' there can be multiple statuses listed in a route/hostname Conditions table.  For example, I think you could have two Rejected, then an Accepted, but need confirmation from @spadgett

spadgett commented:
We should revisit the accepted / rejected logic. We might check the status of the current hostname and show that as the status. Right now we show `Accepted` if any host is admitted, and I'm not certain that's right.

I think there can only be one admitted status per host, however.

dtaylor113 commented:
> We should revisit the accepted / rejected logic. We might check the status of the current hostname and show that as the status. Right now we show Accepted if any host is admitted, and I'm not certain that's right.

yes, for the Status column shown in the Route Listings page.  For Route Details Conditions table we show whatever is in the YAML for the route/hostname.

beanh66 commented:
Okay do we think revisiting that logic is a separate PR @spadgett? Just trying to figure out what we are addressing here. In terms of the visuals, some recommendations below: 
 
@dtaylor113 The mock below includes the Ingress Status header as Sam suggested and just uses spacing to separate the routes if there is more than one. 
<img width="722" alt="screen shot 2018-06-21 at 10 39 08 am" src="https://user-images.githubusercontent.com/18057044/41727202-c8ee723a-7541-11e8-86a3-a1d11b6b84e2.png"> Let me know if this works!

spadgett commented:
Yeah, I’d suggest a separate PR for changing status logic

dtaylor113 commented:
@beanh66, Hi, haven't pushed the code yet, but does this look good to you?

![image](https://user-images.githubusercontent.com/12733153/41745864-6db842d0-7576-11e8-9212-95afde9cfb83.png)

dtaylor113 commented:
Hi @beanh66, I haven't pushed this code yet, but does this look good to you?

![image](https://user-images.githubusercontent.com/12733153/41745921-9daae0ce-7576-11e8-9faf-cb32a908f379.png)

spadgett commented:
Dug into this some more. The sections repeat when there are more than one router. I think we should make "Router my-router" the section header. @beanh66 are you OK with that suggestion?

spadgett commented on frontend/public/components/routes.tsx:
```diff
@@ -267,6 +222,22 @@ const RouteTargetRow = ({route, target}) => <tr>
+const RouteIngressStatus: React.SFC<RouteIngressStatusProps> = ({ingresses}) =>
+  <React.Fragment>
+    {_.map(ingresses, (ingress) =>
+      <React.Fragment key={ingress.host}>
```
Sorry, my previous suggestion was wrong. This should be `key={ingress.routerName}`. `host` is not unique

dtaylor113 replied:
ok, fixed

beanh66 commented:
@spadgett sounds good, so it would look like the following, correct? 

![route](https://user-images.githubusercontent.com/18057044/41773509-48ef514a-75ea-11e8-9c8e-ffd3d5711b9a.png)

dtaylor113 commented:
Here's the latest:

![image](https://user-images.githubusercontent.com/12733153/41779137-dc3144ae-75fe-11e8-87c1-58ad29c62ac3.png)

dtaylor113 commented:
> Dug into this some more. The sections repeat when there are more than one router. I think we should make "Router my-router" the section header.

I'm still not clear on this concept.  We are on a Route Details page, meaning a details page for a single route.  Not sure why there would be two or more routes associated with the single route we drilled down into to see it's details.   Also not sure how to set this scenario up, I programatically did this in the code to mimic two `router.status.ingress`'s.

spadgett commented:
>  Not sure why there would be two or more routes associated with the single route we drilled down into to see it's details.

Routes are exposed by routers. You can have more than one router in a cluster to distribute load. This tells you which of the routers in the cluster have admitted your route. (It's possible to be admitted by one router and rejected by another.)

See https://docs.openshift.org/latest/install_config/router/default_haproxy_router.html#working-with-multiple-routers

dtaylor113 commented:
:+1: Thanks Sam

benjaminapetersen commented on frontend/public/components/_route.scss:
```diff
@@ -15,3 +11,26 @@
 .co-m-route-tls-reveal__btn {
   padding-right: 0;
 }
+
```
BEM? Just validating, there are a lot of `-`, that might be intentional.

dtaylor113 replied:
yeah, I need to ramp up on BEM, I'll check

spadgett replied:
Good catch, it should probably be

```
.co-m-route__ingress-status
.co-m-route__ingress-status__conditions-heading
```

and drop the ingress-status-section class. Wondering how it looks like you just use h2/h3 instead, though? Do we need the custom style here?

dtaylor113 replied:
The custom style is to left-alight the table header and rows so that they appear under the 'Conditions' table header.

dtaylor113 replied:
Updated to BEM names @spadgett suggested.

spadgett commented on frontend/public/components/conditions.tsx:
```diff
@@ -39,7 +38,17 @@ export const Conditions: React.SFC<ConditionsProps> = ({obj}) => {
 /* eslint-disable no-undef */
+export type conditionProps = {
+  type: string;
+  status: boolean | string;
```
Missed this on previous review, but isn't this always a string?

spadgett commented on frontend/public/components/_route.scss:
```diff
@@ -15,3 +11,26 @@
+  .co-m-route__ingress-status {
+    padding-bottom: 40px;
+
+    .co-m-table-grid {
```
This is not really BEM. You shouldn't have this nesting. `.co-m-table-grid` styles shouldn't change based on the page they're on.

spadgett replied:
All of the other tables in the console have some margin. Let's be consistent and not override table styles on just one page. If we want to change it, we should change it everywhere.

For example,

![screen shot 2018-06-22 at 2 23 02 pm](https://user-images.githubusercontent.com/1167259/41792767-f2fec002-7627-11e8-8124-e05914d085d4.png)

spadgett replied:
(Just a quick comment that we are fixing the awkward wrapping in these tables under #149)

dtaylor113 replied:
> All of the other tables in the console have some margin. Let's be consistent and not override table styles on just one page. If we want to change it, we should change it everywhere.

This was a request from @beanh66 , I believe due to the fact that Conditions go along with the current Router and there maybe multiple Routers repeated within the section.   Should each Router actually be it's own section (separated by a section lines)?  If so, then I can see it being like the other sections on the page Ie. "TLS Settings".

spadgett commented:
Hey @dtaylor113 just a small PR with the changes I was thinking of here: https://github.com/dtaylor113/console/pull/1

If @openshift/team-ux-review suggests we remove the left and right padding from tables, I'd like to open an issue to track and make the change everything. Let us know.

dtaylor113 commented:
Hi @spadgett & @beanh66, I feel like we are thrashing on the design here which is causing me to code and recode.  It would be great if we could come to a consensus by cob today as I'm on PTO for the rest of the week :-)  -thanks

spadgett commented:
@dtaylor113 I have a PR against your branch with the changes :)

If we feel there are some tweaks, I believe we can address them in a follow on. I think we agree on the basic design.

dtaylor113 commented:
Hi, here's the latest with @spadgett updates:
![image](https://user-images.githubusercontent.com/12733153/41794307-f65fc75a-762c-11e8-9a7e-9424b2433a96.png)

Only thing I see missing is the 40px space between the repeating content.  @spadgett this is what @beanh66 forwarded to me in slack which you probably weren't aware of: 
![image](https://user-images.githubusercontent.com/12733153/41794392-39dcc6ae-762d-11e8-878b-89b807171885.png)
*Note the 0 left padding/margin for the conditions table as well :-)

spadgett commented:
Let's address any style tweaks in a follow on

spadgett commented:
I don't think the styles are hard to achieve, but I want to make sure we're following BEM. Also I'd like to make sure the styles we use are consistent across pages.

spadgett commented:
/refresh

spadgett commented:
> Pending — Not mergeable. Needs lgtm label.

Hm, tide seems confused

spadgett commented:
40px margin between sections added back in #159
